home/migrations/0099_migrate_content_page_templates_to_standalone_templates.py

```diff
@@ -1,37 +1,105 @@
+    templates = WhatsAppTemplate.objects.all()
+    for template in templates:
+        pages = ContentPage.objects.filter(whatsapp_title=template.name)
+        while pages.count() > 1:
+            page = pages.first()
+            suffix = 1
+            candidate_whatsapp_title = template.name
+            while ContentPage.objects.filter(
+                whatsapp_title=candidate_whatsapp_title
+            ).exists():
+                suffix += 1
+                candidate_whatsapp_title = f"{template.name}-{suffix}"
+            page.whatsapp_title = candidate_whatsapp_title
+            page.save(update_fields=["whatsapp_title"])
 
 
 def migrate_content_page_templates_to_standalone_templates(
-    ContentPage: Any, WhatsAppTemplate: Any
+    ContentPage: Any, WhatsAppTemplate: Any, Image: Any
 ) -> None:
     content_pages = ContentPage.objects.filter(is_whatsapp_template=True)
 
     for content_page in content_pages:
         whatsapp_block = content_page.whatsapp_body[0]
         whatsapp_value = whatsapp_block.value
-        whatsapp_template = WhatsAppTemplate.objects.create(
-            name=content_page.whatsapp_template_name,
-            locale=content_page.locale,
-            message=whatsapp_value.get("message", ""),
-            example_values=whatsapp_value.get("example_values", []),
-            category=content_page.whatsapp_template_category,
-            buttons=whatsapp_value.get("buttons", []),
-            image=whatsapp_value.get("image", None),
-            submission_status="NOT_SUBMITTED_YET",
-            submission_result="",
-        )
-        content_page.whatsapp_body = []
-        content_page.whatsapp_body.append(("Whatsapp_Template", whatsapp_template))
-        content_page.is_whatsapp_template = False
-        content_page.save()
+        print(f"Content Page slug: {content_page.slug}")
+        print(f"Content page title: {content_page.whatsapp_title}")
+        print(f"Whatsapp Value type: {type(whatsapp_value)}, value: {whatsapp_value}")
+        if (
+            not hasattr(whatsapp_value, "_meta")
+            or whatsapp_value._meta.verbose_name != "WhatsApp Template"
+        ):
+            image = whatsapp_value.get("image", None)
+            if image:
+                print(f"Image type: {type(image)}")
+                if isinstance(image, int):
+                    image = Image.objects.get(id=image)
+                elif hasattr(image, "id"):
+                    image = Image.objects.get(id=image.id)
+                elif isinstance(image, dict) and "id" in image:
+                    image = Image.objects.get(id=image["id"])
+                else:
+                    image = None
+            else:
+                print("No image")
+            example_values = list(whatsapp_value.get("example_values", []))
+            whatsapp_template = WhatsAppTemplate.objects.create(
+                name=content_page.whatsapp_title.lower().replace(" ", "_"),
```

Comment:
Instead of the two previous functions, a more robust way would be to have a function here to create a template name, which would do the looping to create a unique template name, and then we pass that value to `WhatsAppTemplate.objects.create`